provider/types.go

```diff
@@ -426,6 +426,67 @@ var extraTypes = map[string]schema.ComplexTypeSpec{
 			{Value: "SecureString"},
 		},
 	},
+	"aws:s3/routingRules:RoutingRule": {
```

Comment:
```suggestion
	"aws:s3/RoutingRules:RoutingRule": {
```

Do these need to all be capitalized like the others?

Reply:
They do not need to be. There is no enforcement and no standard here - do we want to set one to prefer lowercase going forward?

Reply:
I think we should try to keep it consistent to avoid issues with differences in casing like https://github.com/pulumi/pulumi-aws/pull/5497/.

I also realized I thought this was causing an actual issue because none of the SDKs were updated, but it looks like that is because we aren't actually using this type anywhere. It looks like this is for the `aws_s3_bucket_website_configuration` resource which currently has two conflicting properties. `routing_rules` and `routing_rule`. One is already typed and the other is not.

I think we should just get rid of this overlay altogether since there is already a typed version.

Reply:
So this took me a minute to figure out.

The two conflicting versions are actually both from upstream; it is not that one of them stems from the Typescript overlay. That part is fine and works today.

However, `BucketWebsiteConfiguration` isn't actually the source of the dangling reference - the dangling reference comes from `aws_s3_bucket_legacy`, which _references_ the `routing_rules` from BucketWebsiteConfiguration, which only work as full types in TypeScript today. Original PR: https://github.com/pulumi/pulumi-aws/pull/616

This does mean that we can't just remove the overlay - because it was added to ensure that routing rules had strict types. 😿

Reply:
Since we are getting rid of the legacy bucket and only using BucketV2 (renamed to Bucket) in #5479 we might want to instead add an AltTypes that uses the existing `routingRules` type from `BucketWebsiteConfiguration, as long as they are the same.